src/qibolab/pulses/pulse.py

```diff
@@ -80,4 +98,46 @@ def duration(self):
         return 0
 
 
-PulseLike = Union[Pulse, Delay, VirtualZ]
+class Acquisition(_PulseLike):
+    """Acquisition instruction.
+
+    This event instructs the device to acquire samples for the event
+    span.
+
+    Only valid on an acquisition channel.
+    """
+
+    kind: Literal["acquisition"] = "acquisition"
+
+    duration: float
+    """Duration in ns."""
```

Comment:
If this duration is different than the corresponding probe, I am not sure how I would handle it in the instrument. At least for QM, the only way that I am aware to control this duration is the [`smearing`](https://docs.quantum-machines.co/1.2.0/docs/Guides/demod/#timing-of-the-measurement-operation), and maybe also the time of flight, however the latter works like a delay, rather than a different duration. However, we are already these parameters as part of the acquisition config, so wouldn't providing a different duration here duplicate them and potentially even create conflicts? For example a different smearing is needed to force the required acquisition duration, than the one provided in the config.

Reply:
The idea is to use it for instruments where they are truly independent, event per event.

In QM, you should use the `_Readout` object, generated by the `PulseSequence.as_readouts` property.
For the time being, I'm not validating that the probe and acquisition have the same duration, but I'm implicitly assuming so. I could even add an explicit validation in the `_Readout` object, or we can leave this up to the driver (either validate, or neglect one of the two).

Reply:
Using `_Readout` will ignore the `duration` of the probe pulse and use the acquisition one. This may not be ideal without throwing a warning/error if the two durations are different, to tell the users that we are not actually doing this. I have not tried, but it could be that an error will be raised by QM because the waveform will be generated using the probe, while the length (duration) will be read from the acquisition and if the two are not compatible the instrument will probably not like it. This may not be true for rectangular pulses, though, so it may be better to catch this earlier, either in `as_readouts` or in the driver.

Reply:
Warnings are pretty useless in practice (though we could think more about logging), they would simply be ignored. If someone notices a problem, there are high chances we'll be contacted before the user reading the warning...

We could raise manually, either in `as_readouts`/`_Readout` or in the driver. In the first case, it would be imposed to every `as_readouts` user. Though, it is likely that you want it, if you're using `as_readouts`.

Reply:
Generally, I have the impression that separating acquisition makes the life of everyone harder: 
* qibolab developers have to support it in the drivers and maintain nasty functions like `as_readouts`,
* users (or routine authors) have to add more pulses in the sequence to achive the same thing, 
* platform authors have to add an additional pulse in the native measurement, essentially providing only an additional duration, that may even be ignored by some instruments.

and all this needs to be documented and well explained to users. 

It is not clear to me what we get in return, other than the ability to acquire for different duration than the probe or potentially add some delay between the two, which was generally already possible with time of flight/smearing, and the ability to play acquisition without probe which is not clear which instruments can support it and if it is relevant for any practical applications, at least near term.

Reply:
It may not be that useful during qubit operations (for which smearing/time of flight could be enough) but it may be strictly required for a different usage of the devices (i.e. their isolated calibration).

I hoped that smearing/time of flight were doing the exact same thing, just with a more convenient interface for experiments (as a rotation in the space of parameters). If there are instruments for which this is the case, so much the better, it's just a translation. Otherwise, they will fail whatever they can't do.
However, I see the acquisition as a more native operation (I'd argue it is), since the instrument DAC and ADC are different discrete elements, and there should be no bond that makes them operating at the same time. Other than the (programmable) logic built on top.

Reply:
In any case, if it's unbearable, we could even expose the `Readout` as part of the interface. If you don't need the split, you could use that. Possibly, with a simpler constructor (such that you don't have to specify both the events). But it doesn't seem that complicate, to me, defining the acquisition as a separate event (and possibly even avoid time of flight/smearing at all in Qibolab interface, compiling them within the driver).

Even better, but further in the future, we could avoid compiling the measurement completely, just keeping it as an atom in Qibolab representation (even in sequences), since most of the drivers will register the measurement parameters, and then just execute the measurement (which is exactly what smearing and time of flight are doing).